.github/workflows/feature-branch.yml

```diff
@@ -0,0 +1,99 @@
+name: Feature branch CI
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - '**'
+      - '!main'
+      - '!adminV2'
+
+
+env:
+  NODE_VERSION: 20.x
+
+jobs:
+  code-style:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'yarn'
+          cache-dependency-path: 'yarn.lock'
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --immutable
+
+      - name: Code style
+        run: |
+          yarn run prettier:check
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'yarn'
+          cache-dependency-path: 'yarn.lock'
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --immutable
+
+      - name: Lint
+        run: |
+          yarn run lint
+
+  unit-tests:
+    runs-on: ubuntu-latest
+# Enable job when we add tests to the project
+    if: false
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'yarn'
+          cache-dependency-path: 'yarn.lock'
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --immutable
+
+      - name: Unit test
+        run: |
+          yarn run test
+
+  build:
```

Comment:
Can we name it `compile` - maybe it's not the best name, but would be consistent with job from go workflow (easier to remember or eventually automate which steps should be required)

Reply:
In frontend world it is always `build`. That is why vite has also `build` command etc. 

I think that we should not be too consistent across the projects which uses different tech stack. So it would be better to have consistent workflows for GO projects and consistent workflows for frontend projects.

Reply:
It's only a name of the stage, and actually does it really mean that the output you want from this step is the build artifact or rather see if there isn't any compilation errors?

Reply:
It means that project is buildable and in the and as the result you have an artifact. It's not only the compilation. 

So it's missleading. You have compile stage but under the hood you built the artifact.🤷🏻‍♂️